packages/java/endpoint/src/main/java/dev/hilla/crud/filter/Filter.java

```diff
@@ -0,0 +1,13 @@
+package dev.hilla.crud.filter;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "t")
+@JsonSubTypes({ @Type(value = OrFilter.class, name = "o"),
+        @Type(value = AndFilter.class, name = "a"),
+        @Type(value = PropertyStringFilter.class, name = "p") })
```

Comment:
As developers can construct and pass their own filters I would suggest to use more readable names here: `type`, `or`, `and`, `propertyString`.